packages/flutter/lib/src/material/circle_avatar.dart

```diff
@@ -120,7 +120,8 @@ class CircleAvatar extends StatelessWidget {
         child: new MediaQuery(
           // Need to reset the textScaleFactor here so that the
           // text doesn't escape the avatar when the textScaleFactor is large.
-          data: const MediaQueryData(textScaleFactor: 1.0),
+          data: MediaQuery.of(context, nullOk: true)?.copyWith(textScaleFactor: 1.0) ??
```

Comment:
you could just require that there be a media query. Just `[assert(debugCheckHasMediaQuery(context));
](https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/debugCheckHasMediaQuery.html)` at the top of the build method.

Reply:
OK.  So what are the guidelines there: are material widgets expected to work outside of a MaterialApp, or not?

Reply:
Different widgets have different requirements. Many require a Material (not MaterialApp) ancestor, many require a MediaQuery ancestor, many require a Directionality or MaterialLocalizations ancestor. MaterialApp itself isn't required.

Reply:
Yeah, I was aware of the details, I just meant: In general, do we wish to write the widgets in the material/ directory so that they can be used without any of those ambient objects, or do we just expect that people will always use them in the context of a Material Design application.

It sounds like the answer is "We don't expect it to be a problem for people to add the ambient settings themselves if they want to use these widgets independently", which is a perfectly reasonable answer, I just wanted to know how to wire things up. :-)

Reply:
We definitely want it to be possible to use them without MaterialApp, but if people want to not use MaterialApp then yes, they should expect to have to do some of the work that MaterialApp does, which includes introducing a MediaQuery.